homeassistant/components/opower/coordinator.py

```diff
@@ -85,20 +85,50 @@ async def _async_update_data(
         # Because Opower provides historical usage/cost with a delay of a couple of days
         # we need to insert data into statistics.
         await self._insert_statistics()
-        return {forecast.account.utility_account_id: forecast for forecast in forecasts}
+        return {forecast.account.uuid: forecast for forecast in forecasts}
 
     async def _insert_statistics(self) -> None:
         """Insert Opower statistics."""
-        for account in await self.api.async_get_accounts():
-            id_prefix = "_".join(
+        accounts = await self.api.async_get_accounts()
+
+        # Utility account id is not necessarily unique.  For backwards
+        # compatibility, when it is unique, use that in the id and name
+        # generation, but when it is not, use the uuid.
+        utility_account_ids = [account.utility_account_id for account in accounts]
+        duplicate_utility_account_ids = [
+            i for i in set(utility_account_ids) if utility_account_ids.count(i) > 1
+        ]
+
+        id_prefixes = {
+            account.uuid: "_".join(
                 (
                     self.api.utility.subdomain(),
                     account.meter_type.name.lower(),
-                    # Some utilities like AEP have "-" in their account id.
+                    # Some utilities like AEP have "-" in their uuid/utility_account_id.
                     # Replace it with "_" to avoid "Invalid statistic_id"
-                    account.utility_account_id.replace("-", "_"),
+                    account.uuid.replace("-", "_")
```

Comment:
I'd just have these 3 lines stored in a local variable inside the for loop below, revert to the old code without the id_prefixes and name_prefixes dicts, and use this new local variable instead of account.utility_account_id. The code would be more concise and easier to follow.